b+tree.ts

```diff
@@ -1577,6 +1638,59 @@ export class BNodeInternal<K,V> extends BNode<K,V> {
 // has the side effect of scanning the prototype chain.
 var undefVals: any[] = [];
 
+/**
+ * Sums the sizes of the given child nodes.
+ * @param children the child nodes
+ * @returns the total size
+ * @internal
+ */
+export function sumChildSizes<K,V>(children: BNode<K,V>[]): number {
+  var total = 0;
+  for (var i = 0; i < children.length; i++)
+    total += children[i].size();
+  return total;
+}
+
+/**
+ * Determines whether two nodes are overlapping in key range.
+ * @internal
+ */
+export function areOverlapping<K,V>(
```

Comment:
I think this can be simplified to
```
export function areOverlapping<K>(
  aMin: K, aMax: K, bMin: K, bMax: K, cmp: (x: K, y: K) => number
): boolean {
  return cmp(aMin, bMax) <= 0 && cmp(aMax, bMin) >= 0;
}
```